plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/orderstats/OrderStatsRestClient.kt

```
@@ -315,42 +319,47 @@ class OrderStatsRestClient @Inject constructor(
-                                }
-                            }
-                            val payload = FetchNewVisitorStatsResponsePayload(site, granularity, model)
-                            mDispatcher.dispatch(WCStatsActionBuilder.newFetchedNewVisitorStatsAction(payload))
-                        },
-                        { networkError ->
-                            val orderError = networkErrorToOrderError(networkError)
-                            val payload = FetchNewVisitorStatsResponsePayload(orderError, site, granularity)
-                            mDispatcher.dispatch(WCStatsActionBuilder.newFetchedNewVisitorStatsAction(payload))
-                        })
+        println("URL $url")
+        val response = wpComGsonRequestBuilder.syncGetRequest(
+                this,
+                url,
+                params,
+                VisitorStatsApiResponse::class.java
+        )
 
-        request.enableCaching(BaseRequest.DEFAULT_CACHE_LIFETIME)
-        if (force) request.setShouldForceUpdate()
+        return when (response) {
+            is Success -> {
+                response.data?.let {
+                    val model = WCNewVisitorStatsModel().apply {
+                        this.localSiteId = site.id
+                        this.granularity = granularity.toString()
+                        this.fields = it.fields.toString()
+                        this.data = it.data.toString()
+                        this.quantity = quantity.toString()
+                        this.date = date
+                        endDate?.let { this.endDate = it }
+                        startDate?.let {
+                            this.startDate = startDate
+                            this.isCustomField = true
+                        }
+                    }
 
-        add(request)
+                    FetchNewVisitorStatsResponsePayload(site, granularity, model)
+                } ?: FetchNewVisitorStatsResponsePayload(site, granularity)
```

I think this elvis operator is not needed here

```
?: FetchNewVisitorStatsResponsePayload(site, granularity)
```

---

lol. Elvis. Now I can not unsee that.

---

I believe we should return an error not a successful response when the backend returns http 200 with an empty `data`. That's what we do for the other endpoints at least - example https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/a32bc8e71f7991a9a9c33c57573277fd59cff9d0/plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderRestClient.kt#L308. Wdyt?

---

Ohh, we are using [WPComGsonRequestBuilder](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/9876e32b73819d2a091881ca218be9d9b53a487c/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/WPComGsonRequestBuilder.kt#L112) here which has non-nullable `data` field, as oppose to [JetpackTunnelGsonRequestBuilder](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/48069374b1f23056af27bacfbd373267d2515ff6/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/jetpacktunnel/JetpackTunnelGsonRequestBuilder.kt#L179) which has the same field marked as nullable. TIL